### ISSUE TITLE
Api reference with description api

### ISSUE THREAD

**Issue by SmileGobo** [boostorg/signals2#23]
How about api reference with normal description class and methods, as for example in the [stl](http://en.cppreference.com/w/cpp/thread/promise)

**Comment by fmhess** [comment]
I agree the class references are hidden.  Sure, go ahead and make them more prominent.

**Comment by jeking3** [comment]
@SmileGobo any chance you would be able to do what was suggested?

**Comment by SmileGobo** [comment]
@jeking I'm sorry, but I am very busy to do this.
